entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityOperationService.kt

```diff
@@ -0,0 +1,116 @@
+package com.egm.stellio.entity.service
+
+import com.egm.stellio.entity.util.GraphBuilder
+import com.egm.stellio.entity.web.BatchEntityError
+import com.egm.stellio.entity.web.BatchOperationResult
+import com.egm.stellio.shared.model.BadRequestDataException
+import com.egm.stellio.shared.model.ExpandedEntity
+import com.egm.stellio.shared.util.NgsiLdParsingUtils.NGSILD_ENTITY_ID
+import com.egm.stellio.shared.util.NgsiLdParsingUtils.NGSILD_ENTITY_TYPE
+import com.google.common.graph.Graph
+import com.google.common.graph.Graphs.copyOf
+import org.springframework.stereotype.Component
+import kotlin.streams.toList
+
+/**
+ * Service to work on a list of entities.
+ */
+@Component
+class EntityOperationService(
+    private val entityService: EntityService,
+    private val graphBuilder: GraphBuilder
+) {
+
+    /**
+     * Splits [entities] by their existence in the DB.
+     */
+    fun splitEntitiesByExistence(entities: List<ExpandedEntity>): Pair<List<ExpandedEntity>, List<ExpandedEntity>> =
+        entities.partition {
+            entityService.exists(it.id)
+        }
+
+    /**
+     * Creates a batch of [entities].
+     *
+     * @return a [BatchOperationResult]
+     */
+    fun create(entities: List<ExpandedEntity>): BatchOperationResult {
```

Comment:
Doesn't it miss the relationships of properties (we often use this kind of model)?

Reply:
Hmm, no, ok, I was misled by `ExpandedEntity.getRelationships()` (whose name suggests it only retrieves the entity's direct relationships)